### PR TITLE
Add ShareLinks

### PR DESCRIPTION
**Pull request by mackslevin**
Add URL ShareLink in ContentView as Swipe Action and Content Menu action. Add URL ShareLink and note text ShareLink to ellipsis menu in DetailView toolbar.

### CODE REVIEW

File: Memento/Views/DetailView.swift [hunk 1]
```diff
@@ -77,6 +77,19 @@ struct DetailView: View {
...
                         }
                         .buttonStyle(.bordered)
+                        
+                        if let url = item.url {
+                            ShareLink(item: url) {
+                                Label("Share Link", systemImage: "square.and.arrow.up")
+                            }
+                        }
+                        
+                        if let note = item.note, !note.isEmpty {
```
Comment (knotbin) [inline Memento/Views/DetailView.swift]: change to `else if` so two share buttons dont show up if an item has both a note and a link, share should default to link if both are present.

File: Memento/Views/ItemView.swift [hunk 2]
```diff
@@ -78,6 +78,13 @@ struct ItemView: View {
                             UpdateAll()
                         }
                     )
+                    
+                    if let url = item.url {
+                        ShareLink(item: url) {
+                            Label("Share", systemImage: "square.and.arrow.up")
+                        }
+                    }
```
Comment (knotbin) [inline Memento/Views/ItemView.swift:90]: add an `else if` statement to share a note if present

File: Memento/Views/ItemView.swift [hunk 3]
```diff
@@ -98,6 +105,12 @@ struct ItemView: View {
                     UpdateAll()
                 }
             ).tint(.indigo)
+            
+            if let url = item.url {
+                ShareLink(item: url) {
+                    Label("Share", systemImage: "square.and.arrow.up")
+                }
+            }
```
Comment (knotbin) [inline Memento/Views/ItemView.swift:121]: add an `else if` statement to share a note if present

**Review by knotbin — CHANGES_REQUESTED** [review]
Thank you so much for contributing! This means so much for a project I've been working on entirely alone up until this point. The code looks great! I just have a few very minor changes to request.

**Comment by mackslevin** [comment]
Thank you for taking the time to look over my pull request! This is my first time contributing to an open source project; I hope I can be helpful! If I'm understanding right, you'd like in all cases for a URL-based ShareLink if the item URL exists, and otherwise to offer a note text-based ShareLink if possible. I'm implementing this right now and will submit the change shortly. Please let me know if I've misunderstood anything, or if you'd like any further revisions.On Jul 17, 2024, at 2:35 PM, Roscoe Rubin-Rottenberg ***@***.***> wrote:﻿
@knotbin requested changes on this pull request.

Thank you so much for contributing! This means so much for a project I've been working on entirely alone up until this point. The code looks great! I just have a few very minor changes to request.

In Memento/Views/DetailView.swift:
> @@ -77,6 +77,19 @@ struct DetailView: View {
                             }
                         }
                         .buttonStyle(.bordered)
+                        
+                        if let url = item.url {
+                            ShareLink(item: url) {
+                                Label("Share Link", systemImage: "square.and.arrow.up")
+                            }
+                        }
+                        
+                        if let note = item.note, !note.isEmpty {

change to else if so two share buttons dont show up if an item has both a note and a link, share should default to link if both are present.



In Memento/Views/ItemView.swift:
> @@ -78,6 +78,13 @@ struct ItemView: View {
                             UpdateAll()
                         }
                     )
+                    
+                    if let url = item.url {
+                        ShareLink(item: url) {
+                            Label("Share", systemImage: "square.and.arrow.up")
+                        }
+                    }

add an else if statement to share a note if present



In Memento/Views/ItemView.swift:
> @@ -98,6 +105,12 @@ struct ItemView: View {
                     UpdateAll()
                 }
             ).tint(.indigo)
+            
+            if let url = item.url {
+                ShareLink(item: url) {
+                    Label("Share", systemImage: "square.and.arrow.up")
+                }
+            }

add an else if statement to share a note if present

**Comment by knotbin** [comment]
Thank you so much! Super appreciated!

**Review by knotbin — APPROVED** [review]
Looks great! Thank you again!

**Comment by mackslevin** [comment]
So happy to contribute! Thank you.

**Pull request merged**